application-example.conf

```diff
@@ -1,80 +1,130 @@
+    deleteByFieldField: "field_to_delete"
+    deleteByFieldValue: "false"
 }
 
-#alternative syntax for declaring pipelines:
-#pipelines: [{name: "pipeline1", stages: [{class: "com.kmwllc.lucille.stage.CreateChildrenStage"}]}]
-#pipelines: ${pipelines} [{name: "pipeline2", stages: [{class: "com.kmwllc.lucille.stage.CreateChildrenStage"}]}]
-
-dev {
-  kafka {
-    bootstrapServers: "localhost:9092"
-  }
-
-  elastic {
-    host: "localhost",
-    port: "9200"
-    sendEnabled: ${?ELASTIC_SEND_ENABLED}
-  }
-}
-
-aws {
-  accessKeyId: "${?AWS_ACCESS_KEY_ID}",
-  secretAccessKey: "${?AWS_SECRET_ACCESS_KEY}",
-  defaultRegion: "${?AWS_DEFAULT_REGION}"
-}
-
+# elastic search
 elastic {
   url: "http://localhost:9200"
-  index: "lucille-default"
+  index: "index1"
   type: "lucille-type"
   sendEnabled: false
 }
 
-# Solr default mode
+# solr basic indexer config
+# - uses HTTP2SolrClient, url includes the collection name (i.e. "collection1")
+solr {
+  url: "http://localhost:8983/solr/collection1"
+}
+
+# solr cloud mode with url
+# - uses CloudHTTP2SolrClient, url does not provide collection name, instead included as defaultCollection (i.e. "collection2")
+solr {
+  url: ["http://localhost:8983/solr"]
+  useCloudClient: true
+  defaultCollection: "collection2"
+}
+
+# solr cloud mode with zkHosts
 solr {
-  url: "http://localhost:8983/solr/callbacktest"
+  useCloudClient: true
+  defaultCollection: "collection3"
+  zkHosts: ["zookeeper1:2181", "zookeeper2:2181", "zookeeper3:2181"]
+  # optional
+  zkChroot: "/solr"
 }
 
-# Solr cloud mode with url
-# solr {
-#   useCloudClient: true
-#   defaultCollection: "zktest2"
-#   url: ["http://localhost:8983/solr"]
-# }
-
-# Solr cloud mode with zkHosts
-# solr {
-#   useCloudClient: true
-#   defaultCollection: "gettingstarted"
-#   zkHosts: ["zookeeper1:2181", "zookeeper2:2181", "zookeeper3:2181"]
-#   # optional
-#   zkChroot: "/solr"
-# }
+# zookeeper settings for solr indexer
+zookeeper {
+  connectString: "localhost:2181"
+}
 
+# open search
 opensearch {
   url: "https://admin:admin@localhost:9200"
   url: ${?OPENSEARCH_URL} # allow env override
-  index: "lucille-default"
+  index: "index2"
   index: ${?OPENSEARCH_INDEX}
   acceptInvalidCert: false # only enable for testing ssl/https against localhost
 }
 
+#################
+## Worker Config:
+worker {
+  # maximum number of times across all workers that an attempt should be made to process any given document;
+  # when this property is not provided, retries will not be tracked and no limit will be imposed;
+  # this property requires that zookeeper is available at the specified zookeeper.connectString
+  maxRetries: 2
+
+  # number of worker threads to start for each pipeline when running lucille in local / single-JVM mode
+  threads: 2
+
+  # maximum time to spend processing a message, after which the worker will assume a problem and shut down
+  # NOTE: not supported while com.kmwllc.lucille.core.Worker and PipelineWorker are being reconciled
```

Comment:
this "NOTE:" is obsolete & should be removed -- line 119 only

but on line 118 where it says "assume a problem and shut down," need to clarify that shut down only happens if worker.exitOnTimeout=true

as discussed, also want to mention worker.heartbeat and worker.period

Reply:
Addressed in new commit. However, if you could take another look at the comments and explanation I provided, I just want to make sure that it is accurate